### ISSUE TITLE
Proposal: add advertisements to Bitcoin.org

### ISSUE THREAD

**Issue by harding** [bitcoin-dot-org/bitcoin.org#1139]
In PRs #1136, #1087, and #1077, Bitcoin.org domain co-owner @Cobra-Bitcoin has been working towards adding advertisements, and the infrastructure necessary to attract advertisers, to Bitcoin.org.

Several contributors, including myself, have expressed concerns over these plans, and I think that we should have a place to discuss them so that we don't overflow the discussion onto the bitcoin-dev mailing list, where [a thread](http://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-November/011727.html) has already been started.

(Please note that this isn't my proposal; I'm just making an explicit issue for a goal the domain owners are working towards.)

Please see the PRs and the mailing list post above for previous discussion about this issue.

**Comment by benjyz** [comment]
Thanks for handling this in an open and professional manner. I'd challenge the notion that anybody "owns" bitcoin.org. The original owner was Satoshi Nakamoto and all the value clearly is not associated with one person holding the domain. Either it is a public utility, or it is a private enterprise. The owners should decide what they want it to be. If it is to be private it should not further be associated with the bitcoin project. Similar arguments apply to bitcointalk.org.

A counter proposal: bitcoin.org and all "public" bitcoin related domains try to attract capped donations to a pre-defined Bitcoin address. Capped donations would be donations which have a pre-announced amount which are capped to fund hosting e.g. This could avoid the issues bitcointalk had, which received open ended donations and, in parts invalid, I believe, criticism about the use of those funds. 

Advertisements seem like a bad idea for many reasons. Especially Google advertisement is counter to principles of Internet privacy. If the Bitcoin ecosystem can't get together to fund even the hosting of binaries this would be the sign of some deficiency of funding channels. If advertisement it should be not based on Google. It shouldn't be too hard to find say 10 big Bitcoin businesses which sponsor the site.

**Comment by harding** [comment]
This is a reply to @jonasschnelli's [mailing list post](http://lists.linuxfoundation.org/pipermail/bitcoin-dev/2015-November/011727.html)

> Recently, bitcoin.org did merge a PR [1] that enables Google analytics for bitcoin.org. The PR comments did show disagreement for this step. For me, this seems to be against the [about-us](https://bitcoin.org/en/about-us) rules in about "who is in charge of bitcoin".

Please note that the top section of that page is titled, "Who owns bitcoin.org" and it says, "Final publication authority is held by the [domain] co-owners,..."

> What ads would be displayed there?

@Cobra-Bitcoin and @theymos have talked this over with @saivann and myself, and they're going to be vetting all advertisements.  The last plan I heard was to auction Bitcoin.org advertisements on the [BitcoinTalk auctions board](https://bitcointalk.org/index.php?board=73.0) just like the Bitcoin Talk advertisements are auctioned.

> I'd like to know, how changes on bitcoin.org happen? Do they follow consensus-agreement among bitcoin-space contributors or does a group of people decide what to merge and what not?

It is almost always the case that the co-maintainers @saivann, @crwatkins (who focuses on wallets), and myself don't merge until we think we have consensus.  (Although Bitcoin.org's stakes are smaller than Bitcoin Core, so absence of criticism is also considered weak consensus that's good enough for merging, under the expectation that if reasoned criticism appears later, we will revert.)

We do not seek community consensus on issues that don't affect the information or presentation of the site.  For example, @saivann and I often tweak the site build system as necessary without discussing it with anyone besides ourselves, although we do try to make all changes public.

The domain owners retain exclusive control over the domain, so they are technically capable of overriding the maintainers at any time, and they've asserted that they will do that if they feel it's necessary.  As shown above, we make this explicit on our about-us page (it was not explicit when I started contributing).

The two pull requests I can think of (of the top of my head) which were merged with less agreement than I would consider consensus are:

- #894 (contentious hard fork policy, merged by me with the support of the domain owners), see https://github.com/bitcoin-dot-org/bitcoin.org/pull/894#issuecomment-112247780
- #1087 (enable Google Analytics, merged by @Cobra-Bitcoin )

> If site operators or contributers need to get payed for their (highly appreciated) work or need to pay for infrastructure, we should address this root problem.

As of Septemeber, both @saivann and myself are employees of 21 Inc.  @crwatkins is not (as far as I know) affiliated with any Bitcoin company, and our numerous reviewers and less frequent contributors have various affiliations.

The main site cost is the server, which is hosted with Black Lotus who specialize in DDoS resistance.  For that protection, we currently pay $590 a month.  We also have a couple VPSes we use for builds and testing, which cost $10 a month.  Those are our current regular costs.

Thanks to the grant money from the Bitcoin Foundation (and to our contributors for declining to take any stipend money for several months before the grant ended) we are pre-paid on sever expenses through to March 2017 (not a typo, we're prepaid for over a year ahead).

Note, however, that the DDoS-resistant plan we have didn't include enough bandwidth to keep us online [a week ago](https://www.reddit.com/r/Bitcoin/comments/3rumd1/psa_bitcoinorg_is_strongly_ddosed/) when we were attacked, although I think that was the first time we went down in over a year.  One of @Cobra-Bitcoin's concerns is that we may need more money to survive a long-term DDoS at that level.

> I'm pretty sure we can raise funds for a such purpose and I'm offering my help to speak to bitcoin businesses and individuals.

I don't know if the offers are still open, but we've received offers from Bitcoin businesses to entirely replace the $2,000/month we previously received from the Bitcoin Foundation.  Earlier in the year, when it looked like the Foundation was going to be unable to fulfill its commitment, several individuals (including myself) offered to pay the $600/month hosting costs.

I don't think Bitcoin.org lacks friends (but thank you for your offer!); I think the reason we're seeking advertising income is related to this comment from @Cobra-Bitcoin: "My problem with "accepting charity" is that it makes the site dependent on an organization with different goals and priorities from those of the site."

Whether that's good policy or not is what I created this issue to discuss.  Thanks for taking the time to ask the tough questions!

**Comment by jgarzik** [comment]
Must thoroughly screen each ad and company behind it.

Comments and recommendations:

* **User Security**    Malvertising campaigns -- digitally signed! -- hit infosec media headlines at least once every 30-60 days.  InfoSec professionals, I'm guessing, would tell you the true rate of infection is higher than that.

* **Scams**  Above average amount of scams in the bitcoin space.  Must verify a project is legit.

* **Ad networks**  Avoid ad networks.  Adding to the malware risk, ad networks permit customers to directly target specific sites.  Ad network UIs make it trivial to automate targeting bitcoin.org specifically for malware or scams.

* **Analytics** have a far lower malware risk than ad networks, but do carry some remote-code-injection security risks.  Presumably Google Analytics is damn near immune to this... we hope.

**Comment by crwatkins** [comment]
Along with @harding I respect that it is ultimately the domain owner's choice about how this site is funded.  I personally don't believe advertising is the best funding route for this site.

The biggest assets of this site are trust and respect:  Trust in the content which is derived from the respect of the contributors along with the process by which the content is approved.  Display advertising circumvents and weakens this.  Since this is a two way street, not only does advertising weaken my attraction to the site as a user, it drastically weakens my attraction as a contributor.

**Comment by theymos** [comment]
The hosting money has to come from somewhere. (And it'd also be nice to get some extra money to give to significant contributors.) Probably donations would be better, but accepting donations from one big source makes bitcoin.org less independent. I think that crowdfunding might work, though Cobra expressed concerns that this might generate essentially zero income after the first few months. If donations aren't suitable, then I don't have a big problem with ads. (In fact, Wikipedia's refusal to run ads is IMO one of their biggest mistakes, and pretty annoying in light of their intrusive donation drives.)

Maybe putting the ad at the top of the page will make it too prominent. I feel like putting it in the footer might still generate sufficient income while being a lot less intrusive.

The ads will be manually pre-approved by Cobra, and there will be no possible way for advertisers to insert code into bitcoin.org. It isn't possible to entirely eliminate the possibility of scams, or the ability of advertisers to add malware to *their* website. Though if there's a policy of carefully looking at / thinking about advertisers, maybe the risk can be kept very small.

**Comment by harding** [comment]
@theymos can you explain how accepting donations from one big source makes Bitcoin.org less independent?  We accepted all our funds from the Foundation for over a year, and during that time we received funds we were able to apply them to future hosting costs in case the Foundation stopped paying (or, as was never the case, attempted to take away our editorial independence)?

Also, I also have doubts about crowdfunding, but why does that leave the only options left as "one big source" or "advertisements"?  If one big source is not an acceptable option to you and @Cobra-Bitcoin , what about several significant-but-not-majority sources?

In other words, what would be acceptable options to you that don't involve advertising?

As a final question, do you think it would be possible to reduce our hosting costs so that this isn't really an issue?  For example, if the monthly cost was less than $100, it would probably be easy to find ten reliable $10 monthly donors (sign me up).

**Comment by theymos** [comment]
When there is only one or two huge funders, then bitcoin.org contributors will have a conflict of interest toward maintaining the funding arrangement. When the Foundation supported bitcoin.org, it was even very common for people to think that bitcoin.org was *owned* by the Foundation.

Also, if bitcoin.org is receiving only a fixed monthly amount which is supposed to more-or-less cover expenses, then it becomes more difficult to save in order to pay for larger expenses or expansion. We'd have to negotiate with the existing donators (especially if there's an exclusivity agreement) or try to find more donators. With ads, growth seems more natural, since ad revenue will likely grow as the site becomes more popular/useful.

> about several significant-but-not-majority sources?

That'd be fine with me.

As I mentioned in the private emails, I think that it'd be effective to do "gameified" crowdfunding where you pay some relatively amount and then show up randomly in the footer according to a lottery system or something. But that's perhaps a bit complicated to implement.

> As a final question, do you think it would be possible to reduce our hosting costs so that this isn't really an issue?

I don't feel like trying to reduce the scope/quality of bitcoin.org's operations is good in general.

**Comment by harding** [comment]
> When there is only one or two huge funders, then bitcoin.org contributors will have a conflict of interest toward maintaining the funding arrangement.

Why?  Or, rather, it seems to me that the single best thing we can do to ensure our independence is have a nice big hunk of savings.  That's what Saïvann, Craig, and I tried to ensure in the final months of the Bitcoin Foundation sponsorship by declining to accept even the minor compensation we had previously received.  And, look, it's given us something like 15 months of pre-funding.

If we have savings and our funding contracts specify that we can walk away at any time (or with a month's notice at most), why should any of us feel compelled to stay in a bad arrangement?

>  I don't feel like trying to reduce the scope/quality of bitcoin.org's operations is good in general.

Almost every time I tell devs that we pay $600/month to host a static website, they express their shock.  Maybe I just don't understand everything that we offer by being hosted by Black Lotus (and maybe that's a separate conversation), but it does seem rather high to me.

**Comment by luke-jr** [comment]
Does a simple website need proactive DDoS protection anyway? Eligius has had significant success mitigating DDoSs using Amazon EC2 and cheap VPSs on demand. Perhaps @wizkid057 can help advise on reducing costs?

**Comment by xor-freenet** [comment]
Bitcoin is the very definition of software where security is critically 
important.
Hence the main website should by no means deploy *any* kind of unreviewed 
code, even from it's own developer community.
And even worse than that would be deploying arbitrary unreviewed code from 
*strangers*, which ads usually are.
There is no way this can *not* go wrong.

Besides, let's please remember the unfortunate situation of the Bitcoin 
foundation: Hating it has become some kind of meme. Which sucks, because a 
foundation is usually what's necessary to run any kind of big open source 
project.
Do you think displaying ads on sites related to it will help this situation to 
improve?

**Comment by Cobra-Bitcoin** [comment]
@benjyz Domain names have owners who make the final decisions. Bitcoin.org is a private website that serves to educate new users about Bitcoin, it shouldn't be thought of as some sort of vital public utility (that would give it too much significance and importance). Please don't assume Bitcoin.org is "special" just because Satoshi Nakamoto happened to be its owner at one point.

@jgarzik Thank you for your feedback and recommendations!

@harding My view is that the site's long term financial health and ability to expand is best supported by advertising. Ads wouldn't run *all* the time, they would run for short bursts and allow us to dynamically control our savings and budget to meet needs. Exclusive dependence on organizations/donors introduces new risks, such as having to worry about certain changes annoying them, and constantly preparing to find a sponsor if the current one doesn't work out. I'd be surprised if they don't *attempt* to exert some influence on the site's content too. 

We've done this sponsorship thing repeatedly. It always ends with us having to look for a new sponsor after some time. These types of arrangements don't seem well suited in the long term, because sooner or later the sponsor is going to realize they're not getting much out of the arrangement, and then they'll walk away. I can't help but look at the forum and how they fund themselves and think that they've got the right approach.

We're fortunate enough to have enough pre-funding right now to be in a comfortable position. I believe we should use this time to explore alternative funding methods. Nothing is set in stone yet by the way, it's not a certainty that ads will become our main source of funding. I just want the opportunity to try something different here.

@crwatkins The advertising isn't typical banner advertising. It's just a small sentence with a link. There's no pop ups or GIF banners. The ads don't even take up that much space, and they're not going to run 24/7. They're very minimal compared to most advertising on the internet and I think their effect on user trust isn't that much different than the effect a sponsor banner would have.

@luke-jr We've had quite a few DDoS attacks over the years. Maybe we could explore those DDoS mitigation techniques if we ever move back to regular non-DDoS protected hosting (which may not be for quite a while because we've got our hosting prepaid for 15 months).

@xor-freenet The ads themselves don't run any code. You're free to review the code that will display ads to better understand how they're shown: https://github.com/bitcoin-dot-org/bitcoin.org/pull/1136.

**Comment by harding** [comment]
@Cobra-Bitcoin that was an excellent answer; thank you very much for taking the time to write it up.  I'm going to think on this overnight before replying.  Thank you to everyone else who commented as well.

**Comment by gmaxwell** [comment]
> (In fact, Wikipedia's refusal to run ads is IMO one of their biggest mistakes, and pretty annoying in light of their intrusive donation drives.)

This is a subject that I have direct personal experience (having been involved in running the wikimedia fundraisers in 2006-2008)--  Wikimedia's fundraising today is able to bring in drastically more income than the organization is able to effectively spend; and the process has been handled historically in such a way as to avoid building an overlarge warchest because of a principled belief that the service should continue to serve the public interest, and that if it is unable to do so enough to bring in sustaining revenue it should fail and be replaced with something that can. Effectively, the reliance on public support is an intentional and virtuous check on Wikimedia's behavior.

One way to handle large donations from diversified sources is to intentionally direct those funds to contingency or special projects instead of operations, so that the service does not become dependent on them; this was a mechanism which I promoted and was ultimately employed at Wikimedia ( https://wikimediafoundation.org/wiki/Resolution:Update_of_Gift_Policy_-_January_2008 ).  This only works, however, if sufficient diverse sources of income exist to begin with.

That said--  I'm not opposed to having ads which are handled _well_-- e.g. that are unintrusive, are clearly ads, are minimal useful for promoting scams or malware, and do not compromise user privacy.  Think more along the lines of the sponsorship statements on public radio in the US.  The latter point is perhaps a bit sticking because already ads have already driven the use of third party analytics which objectively do reduce user privacy (though perhaps not more than the use of DOS mitigation).  I think they could be part of a sustainable support model which should also include other forms of support; but similar to large donations the potential for distorting influence from advertisements should not be disregarded.

**Comment by saivann** [comment]
>"It always ends with us having to look for a new sponsor after some time"

@Cobra-Bitcoin I don't really understand that argument, as dealing with ads auctions (and doing it well) is similar process, without interruption. With one or many sponsors, you go into this process once every few months at worse, every few years at best.

Either way, I am less reluctant to the idea of just "trying" ads, as opposed to seriously adopting them. What I am more uncomfortable with is that the project has thus far gone in that direction without real negociation (e.g. suggesting the trial period like you just did and offering contributors to have a say on the result as a compromise). I feel like this is very different than how I, or @harding have been working previously and I also share concerns that this risks turning contributors away.

**Comment by benjyz** [comment]
"Domain names have owners who make the final decisions. Bitcoin.org is a private website that serves to educate new users about Bitcoin, it shouldn't be thought of as some sort of vital public utility (that would give it too much significance and importance). Please don't assume Bitcoin.org is "special" just because Satoshi Nakamoto happened to be its owner at one point."

This reads as: I don't have any obligations. AFAIK Nakamoto handed over different Bitcoin assets to different parties, most importantly the keys to the sourcecode, which have been handled with great care. Bitcoin.org is very clearly related to bitcoin core, and this statement leads me to wonder how you come to be owner of this domain. Did you buy it and you should receive all possible future income, including the ability to sell it? Its probably worth 1M$+. Clearly, bitcoin.org is everything but a normal domain.

**Comment by Cobra-Bitcoin** [comment]
@gmaxwell Thank you for sharing your thoughts and feedback! Your comment was very enlightening, especially with regards to the reasoning behind Wikimedia's aggressive fundraising.

@saivann It was always my intention to have a "trial" period and test out ads and see what kind of reaction they get, what type of revenue they bring, and how the community react to them. I feel we're better equipped to decide the right funding approach (long term) once we've experimented with all the different options (donations will also be looked into).

@benjyz I don't see how it contradicts anything. You seem uncomfortable with the idea of private ownership over Bitcoin.org. But this has always been the case, and will continue to be, because of how DNS works. If Bitcoin.org doesn't do a good job of meeting the community's needs, then the community will use alternative websites (like Bitcoin.com or WeUseCoins.com). You're assigning too much importance to one domain and trying to make Bitcoin appear weaker and more centralized than it actually is. This is also the case with Bitcoin Core by the way. Ultimately the community and free market decide what software to run and what websites to promote/use.

**Comment by benjyz** [comment]
I know how DNS works. Its perfectly possible to transfer this property to a non-profit and make a charter around it. If you don't know this stuff works, consult with people who do, because there is plenty of expertise in the Bitcoin community. The big difference with Bitcoin.com and any other bitcoin related domain is that the current owners have received the domain as a gift by Satoshi Nakamoto (whoever he is). This has nothing to do with the free market.

**Comment by harding** [comment]
I really liked what @gmaxwell said here:

> the [Wikimedia Foundation fundraising] process has been handled historically in such a way as to avoid building an overlarge warchest because of a principled belief that the service should continue to serve the public interest, and that if it is unable to do so enough to bring in sustaining revenue it should fail and be replaced with something that can.

That sounds to me like a reasonable model for Bitcoin.org.  If we can't bring in an average of $600/month in sponsorships and donations (the amount necessary to fund the current servers), we should reduce the scope of our operations to the level of donations we do receive.[1]  Of course, if we bring in more, we can add services or pay people to do the stuff that isn't fun.

I think that if we make it clear how decisions are made on the site---which is what we did when the Foundation was sponsoring us---then it will be easy to say "no" to any donor or sponsor who tries to bypass that process.  And if that means losing their future pecuniary contributions, then we'll just have to reduce the scope of our operations to the level supported by the remaining sponsors and donors.

This sounds a lot better to me from a community standpoint than advertising, and it also has the advantages of eliminating the concerns over misleading ads and privacy-destroying user tracking.

[1] I would also still like to explore ways to reduce server costs no matter what happens.

**Comment by xor-freenet** [comment]
> @xor-freenet The ads themselves don't run any code. You're free to review the code that will display ads to better understand how they're shown: #1136.

Let's see:
```javascript
+		var ad_slot_img = document.createElement("img");
+		ad_slot_img.src = "/img/ads/" + ad.icon + ".png";
```

Now check this:
http://www.libpng.org/pub/png/libpng.html

> Security and Crash Bugs in Older Versions
> [...] All "modern" versions of libpng [...] fail to correctly handle malloc() failure for text chunks (in png_set_text_2()), which can lead to memory corruption and the possibility of execution of hostile code. [...]

Fact is: The more bits you pull from third parties, the higher the risk. Doesn't matter whether you intend them to be code, or whether they just maliciously execute as code.

**Comment by Cobra-Bitcoin** [comment]
@benjyz My point is that that difference is irrelevant. It doesn't make the site special, nor does it compel Bitcoin.org's owners to run the site a certain way. 

@harding That definitely sounds like a model we could explore.

@xor-freenet Thank you for your feedback! I must admit I hadn't thought of that risk. It's definitely something we'll think about before experimenting with ads. Maybe the ads could be text-only or something?

**Comment by xor-freenet** [comment]
> @xor-freenet Thank you for your feedback! I must admit I hadn't thought of that risk. It's definitely something we'll think about before experimenting with ads. Maybe the ads could be text-only or something?

Text-only would fix that, yes. Albeit new ads must not be deployed automatically then, but manually reviewed first: They could be abused for text which talks to the user as if the text were instructions of the regular site. This could then tell the user to do unsafe thngs.

However, how about you realize this as an option to get money before you go down the shady path of advertising:
Bitcoin as a project is *perfect* for doing a donation fundraiser such as Wikipedia does. This is because Bitcoin is a payment system. So while the users are at your website, they already have the "money in their hands" by running the payment software anyway.
So do a fundraiser, and only consider advertising if it completely fails.

**Comment by ABISprotocol** [comment]
NACK.  Plus I use adblocker (in my case, Privacy Badger from EFF, and NoScript).

**Comment by luke-jr** [comment]
I wonder if just the "sponsored by" link at the bottom (like BCF used to get) would be "advertising" enough to sustain the site?

**Comment by ABISprotocol** [comment]
I wouldn't object to that.

**Comment by fraggle222** [comment]
This is a really slippery slope. Once ads are allowed in one way, that will inevitably expand. Soon we will have people paying to have their bitcoin wallet promoted as a 'recommended' wallet. 

If bitcoin.org can't afford $600/month then we have problems. 

Would @Cobra-Bitcoin (and other owners) be willing to give back any money raised over cost of running the site to the community somehow?

**Comment by Azulan** [comment]
NACK

Advertisements are a vector for malware. Bitcoin users are a huge target.

**Comment by Azulan** [comment]
@crwatkins  "The biggest assets of this site are trust and respect: Trust in the content which is derived from the respect of the contributors along with the process by which the content is approved."

This was all lost when bitcoin.org showed its true self and delisted a significant and legitimate wallet for petty political reasons.

**Comment by SartoNess** [comment]
who will control the funds.? if they go to a non-profit foundation may be accepted,for bitcoin.org maintenance can get funded from bitcointalk ads funds

**Comment by ABISprotocol** [comment]
My initial observations on this led to me giving this a NACK but then after looking at some more commentary by @luke-jr who wondered if it would be good with basically just a "sponsored by" link at the bottom (like BCF used to get), then I felt I would be o.k. with that.  However, I do have some final concerns:
1) The ads that are run should not be "counter to" the actual wallet selection criteria.  There should be some flexibility for discretion in terms of the actual ad(s) to be run but basically it should be that here there wouldn't be things advertised that wouldn't pass muster if you had to run them by the [wallet selection criteria](https://github.com/bitcoin-dot-org/bitcoin.org#wallets).

2) Disclose the funds raised by the ads and provide a periodic accounting of where they go, basic transparency.

**Comment by mine2marz** [comment]
Bitcoin is "special" not because of Satoshi Nakamura but everything to do with how unique it is. Dressing Bitcoin in ads is an easy way to eliminate finding sponsors often but don't forget why people choose Bitcoin. For its security and privacy. But also because it's not mainstream.

**Comment by i-rme** [comment]
Bitcoin.org website should be hosted in a CDN like Cloudflare or even in github pages, that could solve the DDoS issues.

**Comment by JustAHappyKid** [comment]
Not sure if this is of any value, but Bitcoin.org could potentially earn a little revenue by [adding a referral code](https://spendabit.co/search-partners) to the Spendabit link found on the [Resources page](https://bitcoin.org/en/resources). Perhaps other services/businesses have similar programs.

The biggest problem with this might be the overhead of managing the referral accounts/funds, though in our (Spendabit's) case we'd be happy to consider making automated payouts.

**Comment by harding** [comment]
An advertisement was added in #2485, so closing this discussion issue.
